### PR TITLE
Forwarding progress stream changes from Foreach-Object -Parallel runspaces

### LINKED ISSUE #14254: foreach -Parallel does not show up the progress bar
## Steps to reproduce
I try to copy file to multiple computers with:
```
PS C:\Work> $anps | foreach -Parallel {Copy-Item -Path C:\Install\AN\alila20200117.exe -ToSession $_ -Destination c:\software -Verbose} -Verbose

```

## Expected behavior
I hope it will show up the progress bar, just like what it will do in PSVersion 5.1.18362.1171
image

## Actual behavior

It just show me it's copying to which computer, without progress bar.
```
PS C:\Work> $anps | foreach -Parallel {Copy-Item -Path C:\Install\AN\alilang20200117.exe -ToSession $_ -Destination c:\software -Verbose} -Verbose
VERBOSE: Performing the operation "Copy File" on target "Item: C:\Install\AN\alila20200117.exe Destination: c:\software".
VERBOSE: Performing the operation "Copy File" on target "Item: C:\Install\AN\alila20200117.exe Destination: c:\software".
VERBOSE: Performing the operation "Copy File" on target "Item: C:\Install\AN\alila20200117.exe Destination: c:\software".
VERBOSE: Performing the operation "Copy File" on target "Item: C:\Install\AN\alila20200117.exe Destination: c:\software".
```

## Environment data

```PS C:\Work> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      7.1.0
PSEdition                      Core
GitCommitId                    7.1.0
OS                             Microsoft Windows 6.3.9600
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0

```

### PR DESCRIPTION
**Pull request by powercode**
# PR Summary

Forwarding progress stream changes from `Foreach-Object -Parallel` runspaces in the same way as the other streams.
#14254
## PR Context

Now looks like this.
That it doesn't render perfectly is a separate issue.

```powershell
function Show-Progress {
    param([int] $ThrottleLimit = 4)
    $parentId = 100
    Write-Progress -Id $parentId -Activity Parent -status Parallel
    1..20 | Foreach-Object -ThrottleLimit $ThrottleLimit -Parallel {
        $progress = @{
            Id               = $_
            Activity         = "Activity $_"
            CurrentOperation = "Operation $_"
            PercentComplete  = 0
            ParentId         = $using:parentId
        }
        
        Write-Progress @progress
        Start-sleep -Seconds 1
        $id
        foreach ($p in 1..100) {
            $progress.PercentComplete = $p
            Write-Progress @progress
            Start-Sleep -Milliseconds 100
        }
        start-sleep -Seconds 1
        Write-Progress @progress -Completed
    }         
    Write-Progress -Id 1 -Acti Parent -completed
}
```

image

image

The test looks different as the Progress stream is not forwardable. So the API is used instead.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary. This may include:
        - Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode (which runs in a different PS Host).
        - Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
        - Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
        - Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).

### CODE REVIEW

**Comment by iSazonov** [comment]
@powercode Please add some example scripts/screenshot. 
And we need test(s).

**Comment by tlsalex** [comment]
A few days had passed , the fix still on its way ?

File: test/powershell/Modules/Microsoft.PowerShell.Utility/Foreach-Object-Parallel.Tests.ps1 [hunk 1]
```diff
@@ -77,6 +77,19 @@ Describe 'ForEach-Object -Parallel Basic Tests' -Tags 'CI' {
...
+    It 'Verifies progress data streaming' {
+
+        $ps = [Powershell]::Create([System.Management.Automation.RunspaceMode]::NewRunspace)
+
+        $ps.Commands.AddScript("`$ProgressPreference='Continue'; 1..2 | ForEach-Object -Parallel { Write-Progress -Activity Progress -Status Running }") | Out-Null
+        $ps.Invoke()
+        $pr = $ps.Streams.Progress.ReadAll()
+        $pr.Count | Should -be 2
+        $pr[0].Activity | Should -be Progress
+        $pr[0].StatusDescription | Should -be Running
```
Comment (iSazonov) [inline test/powershell/Modules/Microsoft.PowerShell.Utility/Foreach-Object-Parallel.Tests.ps1:89]: If Count is 2 why do we check only first element?
```suggestion
        $pr.Count | Should -Be 2
        $pr[0].Activity | Should -Be Progress
        $pr[0].StatusDescription | Should -Be Running
```

**Review by PaulHigin — APPROVED** [review]
LGTM  @powercode Thanks for adding this!

**Review by iSazonov — APPROVED** [review]

**Comment by tlsalex** [comment]
`34 successful and 1 expected checks`

what action need to take for the 1 "expected checks" ?

**Comment by tlsalex** [comment]
@powercode could you help to compile latest Powershell code but with your fix for windows x64 ? I want to use this feature before official release .  I am not a code man , I don't know how to compile it. thanks !

**Comment by iSazonov** [comment]
@tlsalex You can download build.zip from CIs.

**Pull request merged**